nomad/structs/structs.go

```diff
@@ -3817,6 +3820,129 @@ type TaskEvent struct {
 
 	// GenericSource is the source of a message.
 	GenericSource string
+
+	// DisplayMessage is a human friendly message about the event
+	DisplayMessage string
+
+	// Details is a map with annotated info about the event
+	Details map[string]string
+}
+
+func (event *TaskEvent) PopulateEventDisplayMessage() {
```

Comment:
Maybe have a couple basic tests for a sanity check.

Reply:
Done